### PR TITLE
Change a logic in pipeline test regarding TF

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

- **Currently**, the tiny models in pipeline tests are created using `model_class(config)`, and for TF models, the weights are not created at this point. Then we set the device (**which turns to be CPU instead of GPU!**), and the wegiths are created in CPU context --> we get the expected exceptions.
- In the future, we will use the tiny model from Hub repos
- The models are loaded using `from_pretrained`
- If GPU available, those weights are initialized in GPU context automatically for TF models, including embedding layers
- From what @Rocketknight1 says at the end, we won't get the expected exceptions in this situation (TF, layer weights loaded in GPU context)

**In order to use the tiny models from the Hub without any pipeline test failure, we will have to skip this check under the above described situation.**

From @Rocketknight1 
> Embedding layers in Keras have different behaviours on CPU and GPU when you pass invalid indices. On CPU, the layer checks inputs and throws an error if you're out of range, but on GPU you just get a zeros tensor as output with no error

### CODE REVIEW

File: tests/pipelines/test_pipelines_summarization.py [hunk 1]
```diff
@@ -64,8 +65,15 @@ def run_pipeline_test(self, summarizer, _):
         if model.config.__class__.__name__ not in model_can_handle_longer_seq:
             # Switch Transformers, LED, T5, LongT5 can handle it.
             # Too long.
-            with self.assertRaises(Exception):
-                outputs = summarizer("This " * 1000)
+            # For TF models, if the weights are initialized in GPU context, we won't get expected index error from
+            # the embedding layer.
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_summarization.py:69]: This is necessary if we want to use tiny models from the Hub for pipeline tests.

File: tests/pipelines/test_pipelines_summarization.py [hunk 2]
```diff
@@ -64,8 +65,15 @@ def run_pipeline_test(self, summarizer, _):
...
+            if not (
+                isinstance(model, TFPreTrainedModel)
+                and get_gpu_count() > 0
+                and len(summarizer.model.trainable_weights) > 0
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_summarization.py:73]: At the moment on `main`, this condition `len(summarizer.model.trainable_weights)` is `False`, as the model is not obtained from `from_pretrained` but by `model_class(config)` which won't create any TF weight at this point.

Comment (sgugger) [reply tests/pipelines/test_pipelines_summarization.py:73]: So should the model be properly created with weights (guessing it needs a forward pass on dummy inputs)?

Comment (ydshieh) [reply tests/pipelines/test_pipelines_summarization.py:73]: @sgugger  On current `main`, there is no test failure around what I mentioned in this PR. We don't get the weights at line, and **they will be created inside pipeline forward** (which is being in **CPU context**, despite we have GPU), and we get the expected exception --> not test failure.

This PR is just to make the necessary changes in order my WIP PR could be merged without failure, where `len(summarizer.model.trainable_weights) > 0` will become `True` --> and we need to skip in this case.

Comment (sgugger) [reply tests/pipelines/test_pipelines_summarization.py:73]: I still don't get why it's useful to support a model with no weights here. The fix is to make sure they have weights.

**Review by sgugger — APPROVED** [review]
ok, thanks for the explanations!

**Comment by Rocketknight1** [comment]
cc @gante to this one - he's been working on a way to check for invalid indices even for embedding layers on GPU

**Review by Narsil — APPROVED** [review]
LGTM

**Pull request merged**